gf-webapp/src/logic/bookingPage/form/Form.tsx

```diff
@@ -1,53 +1,46 @@
-import {useRef, useState} from "react";
+import React, {useState, useRef} from "react";
 import {useModalVisibilityContext, useDictionary} from "../../DictionaryContext/useDictionary";
-import {send} from "emailjs-com";
+import {MailApiService} from "src/service/MailApi/MailApi";
+import {ApiBookingRequestPostRequest} from "src/autogenerated/gf-server-api/models/ApiBookingRequestPostRequest";
 import {Button} from "gf-ui-lib/components/Button/Button";
 import styles from "./Form.module.scss";
 
 type RoomProps = {
   roomNumber: string | null;
 }
 
+const defaultBookingFormData: ApiBookingRequestPostRequest = {
+  email: "",
+  name: "",
+  dateIn: "",
+  dateOut: "",
+  roomNumber: 0,
+  adultsAmount: 0,
+  childrenAmount: 0,
+};
+
 export const Form = (props: RoomProps) => {
   const {bookingRoomForm} = useDictionary().dictionary;
   const {isModalActive, setIsModalActive} = useModalVisibilityContext();
+  const [bookingFormData, setBookingFormData] = useState<ApiBookingRequestPostRequest>(defaultBookingFormData);
 
-  const [toSend, setToSend] = useState({
-    // from_name: "",
-    to_name: "",
-    // message: "",
-    reply_to: "",
-    check_in: "",
-    check_out: "",
-    room_number: "",
-    adults_amount: "",
-    children_amount: "",
-  });
+  const sendMail = async () => {
+    const res = await MailApiService.bookingMessage({apiBookingRequestPostRequest: bookingFormData});
+  };
 
-  const onSubmit = (e: React.ChangeEvent<HTMLFormElement>) => {
+  const onSubmit = async (e: React.ChangeEvent<HTMLFormElement>) => {
     e.preventDefault();
-    send(
-      "service_q2x5wyb",
-      "template_p5netbs",
-      toSend,
-      "Iz-maWBTKYHMeQmxB",
-    )
-      .then((response) => {
-        console.log("SUCCESS!", response.status, response.text);
-      })
-      .catch((err) => {
-        console.log("FAILED...", err);
-      });
-    e.target.reset();
+    setBookingFormData(defaultBookingFormData);
```

Comment:
what happened here? why do we need to set default booking form data?

Reply:
Deleted set default values in inputs